app/src/main/java/com/github/se/travelpouch/model/activity/ActivityRepositoryFirebase.kt

```diff
@@ -0,0 +1,160 @@
+     *
+     * @param onSuccess (List<Event>) -> Unit): The function called when the retrieving of the events
+     *   went successfully.
+     * @param onFailure ((Exception) -> Unit): The function called when the retrieving of the events
+     *   fails.
+     */
+    override fun getActivity(onSuccess: (List<Activity>) -> Unit, onFailure: (Exception) -> Unit) {
+        Log.d("EventRepository", "getEvents")
+        db.collection(collectionPath)
+            .get()
+            .addOnSuccessListener { result ->
+                val events = result?.mapNotNull { documentToEvent(it) } ?: emptyList()
+                onSuccess(events)
+            }
+            .addOnFailureListener { e ->
+                Log.e("EventRepository", "Error getting documents", e)
+                onFailure(e)
+            }
+    }
+
+    /**
+     * This function adds an event to the collection of events in Firebase.
+     *
+     * @param event (Event) : the event we want to add on Firebase
+     * @param onSuccess (() -> Unit) : the function called when the event is correctly added to the
+     *   database
+     * @param onFailure ((Exception) -> Unit) : the function called when an error occurs during the
+     *   adding an event to the database
+     */
+    override fun addActivity(
+        activity: Activity,
+        onSuccess: () -> Unit,
+        onFailure: (Exception) -> Unit
+    ) {
+        performFirestoreOperation(
+            db.collection(collectionPath).document(activity.uid).set(activity), onSuccess, onFailure)
+    }
+
+    /**
+     * This function is a helper function that safely performs a Firebase operation. A task has
+     * listeners added to it. If the task is successful, we apply onSuccess. Otherwise we perform
+     * onFailure.
+     *
+     * @param task (Task<Void>) : a task to perform
+     * @param onSuccess (() -> Unit) : the function called when the event is correctly added to the
+     *   database
+     * @param onFailure ((Exception) -> Unit) : the function called when an error occurs during the
+     *   adding an event to the database
+     */
+    private fun performFirestoreOperation(
+        task: Task<Void>,
+        onSuccess: () -> Unit,
+        onFailure: (Exception) -> Unit
+    ) {
+        task
+            .addOnSuccessListener { onSuccess() }
+            .addOnFailureListener { e ->
+                Log.e("EventRepositoryFirestore", "Error performing Firestore operation", e)
+                onFailure(e)
+            }
+    }
+
+    /**
+     * This function converts a document got from Firebase to an event. It returns null if an error
+     * occurs.
+     *
+     * @param document (DocumentSnapshot) : The document from Firebase
+     * @return (Event?) : If the conversion goes without a problem an event is return. Otherwise, null
+     *   is returned.
+     */
+    private fun documentToEvent(document: DocumentSnapshot): Activity? {
```

Comment:
change documentToEvent to documentToActivity

Reply:
My bad. Copy pasting hazards XD.